### ISSUE TITLE
[FEATURE REQUEST] label_smoothing

### ISSUE THREAD

**Issue by bagustris** [bagustris/nkululeko#10]
**Why do you propose this feature?**
Label smoothing can improve classification performance, particularly on unseen (test) data, maybe for MLP only, but can be generalized to other models too.

Adapting from here maybe the fastest one: https://github.com/human-ai-lab/APSIPA2025GC-ASC/blob/e5801767cb02fd952c2f714f122beb014b640510/improvements.py

**Describe the solution you'd like**
```ini
[MODEL]
# using cross-entropy
loss = ce
label_smoothing = True
```
